modules/images/webp-uploads/load.php

```diff
@@ -573,3 +575,43 @@ function webp_uploads_update_featured_image( $html, $post_id, $attachment_id ) {
 	return webp_uploads_img_tag_update_mime_type( $html, 'post_thumbnail_html', $attachment_id );
 }
 add_filter( 'post_thumbnail_html', 'webp_uploads_update_featured_image', 10, 3 );
+
+/**
+ * Returns an array of image sizes that can have alternative mime types. By default, only core sizes
+ * are allowed and will be returned.
+ *
+ * If a custom size wants to have additional mime types support, they will need to have
+ * the provide_additional_mime_types property in the size details or it should be added using the
+ * webp_uploads_image_sizes_with_additional_mime_type_support filter.
+ *
+ * @since n.e.x.t
+ *
+ * @return array An array of image sizes that can have additional mime types.
+ */
+function webp_uploads_get_image_sizes_with_additional_mime_type_support() {
+	$additional_sizes = wp_get_additional_image_sizes();
+	$allowed_sizes    = array(
+		'thumbnail'      => 'thumbnail',
+		'medium'         => 'medium',
+		'medium_large'   => 'medium_large',
+		'large'          => 'large',
+		'post-thumbnail' => 'post-thumbnail',
+	);
```

Comment:
Using a map here doesn't really make sense I think. This is documented as a simple array (as in "indexed") of image sizes, and having it be a map makes using the filter more complicated.

Maybe you're doing this to be able to use `isset` instead of `array_search`? That seems reasonable to me, but we should handle that internally so that external developers don't have to deal with it.

I'd say either we should handle the transformation to this "map" in a way that the filter usage is not affected by it, or if we prefer to use a map it should avoid the need to define duplicate values like now. If we wanted this to be a map for more efficient PHP lookups, I think the better approach would be to include all sizes by default as keys and then use a boolean as value, which would only be `true` for the core sizes.

If we do the latter, we also will need to update all documentation for the return value and most importantly filter value to clarify that format.

Reply:
Thanks, @felixarntz. I have left the map, but updated it to have size names only as keys and used a boolean as values as you said. Does it look good now?